istioctl/pkg/kubernetes/client.go

```diff
@@ -217,3 +220,62 @@ func (client *Client) GetPilotAgentContainer(podName, podNamespace string) (stri
 	}
 	return proxyContainer, nil
 }
+
+type podDetail struct {
+	binary    string
+	container string
+}
+
+// GetIstioVersions gets the version for each Istio component
+func (client *Client) GetIstioVersions(namespace string, short bool) (string, error) {
+	pods, err := client.GetIstioPods(namespace, map[string]string{"labelSelector": "istio"})
+	if err != nil {
+		return "", err
+	}
+	if len(pods) == 0 {
+		return "", errors.New("unable to find any Istio pod in namespace " + namespace)
+	}
+
+	labelToPodDetail := map[string]podDetail{
+		"pilot":            {"/usr/local/bin/pilot-discovery", "discovery"},
+		"citadel":          {"/usr/local/bin/istio_ca", "citadel"},
+		"egressgateway":    {"/usr/local/bin/pilot-agent", "egressgateway"},
+		"galley":           {"/usr/local/bin/galley", "validator"},
+		"ingressgateway":   {"/usr/local/bin/pilot-agent", "ingressgateway"},
```

Comment:
I don't think we should include gateways. People are likely to deploy their own..

Reply:
The filter I'm doing is getting all pods with `istio` label. If one doesn't deploy a gateway, it won't appear in the list.